MonamiYP/Unmasked
# UM-7_entity component system

### What has been done

- Created an entity component framework to manage entities and components more effectively
- Repurposing PositionComponent to TransformComponent to include rotations and scaling in the future
- Created SpriteComponent to manage the display of the sprites
- Created a Vector2D class to make use of operator overloading to make vector arithmetic easier

### Tickets
- [UM-7 (ECS)](https://monamiyp.atlassian.net/jira/software/projects/UM/boards/4?selectedIssue=UM-7)
- [UM-12 Vector2D](https://monamiyp.atlassian.net/jira/software/projects/UM/boards/4?selectedIssue=UM-12)
- [UM-13 Sprite Component](https://monamiyp.atlassian.net/jira/software/projects/UM/boards/4?selectedIssue=UM-13)
- [UM-14 Transform Component](https://monamiyp.atlassian.net/jira/software/projects/UM/boards/4?selectedIssue=UM-14)

### How to test new features

- You can try out the Vector2D class by editing line 61 in game.cpp to use any of the operators
- Everything else is the same on the surface, so just make sure nothing is broken lol
